tensorflow/core/kernels/mkl_softmax_op.cc

```diff
@@ -81,13 +92,22 @@ class MklSoftmaxOp : public OpKernel {
           layout_type = memory::format::tnc;
           break;
         case 4:
-          layout_type = memory::format::nchw;
+          if (src_mkl_shape.IsMklTensor()) {
+            layout_type = memory::format::nhwc;
```

Comment:
I thought MKL tensors are in NCHW format? Could you please add comments on why the layout is set this way (in the file)?

Also, why does layout matter in softmax? (NHWC vs NCHW, etc.)

Reply:
Thanks, I will add comments.
When input is MKL tensor, layout_type here is used for output tensor. When input is MKL tensor, output is also MKL tensor, but we need to set its TF layout_type.  TF layout is NHWC.
When input is TF tensor, layout_type here is no special sense. We use axis to define on which dimension to do softmax.

Reply:
Thank you for the explanation! This PR should be good to go after the comments.